### DISCUSSION TITLE
Issues installation and first run firedrake

### DISCUSSION THREAD

**Discussion by AntoTir** [firedrakeproject/firedrake#3578] [Firedrake support]
Hello everyone,
i am trying installing Firedrake on my pc with Ubuntu. I complete the installation without any problem, the  following tests

> cd $VIRTUAL_ENV/src/firedrake
>pytest tests/regression/ -k "poisson_strong or stokes_mini or dg_advection"

are passed, with just these warnings

=============================== warnings summary ===============================
../fiat/FIAT/__init__.py:5
  /home/raffaele/firedrake/src/fiat/FIAT/__init__.py:5: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========== 27 passed, 4 skipped, 3507 deselected, 1 warning in 58.61s ==========
WARNING! There are options you set that were not used!
WARNING! could be spelling mistake, etc!
There is one unused database option. It is:
Option left: name:-k value: poisson_strong or stokes_mini or dg_advection source: command line

Then, I use it in vs code for the tutorial notebook. By the way, the plots are always wrong (looking at the plot reported into the notebook).. I have tried to use it in google colab, and there the same notebook that gives problem on my pc, produces reasonable plot. I really don't get where is the problem.

PS: the command
>%matplotlib widget

is giving a lot of problem. I have tried all the different solutions proposed on different forums, but they do not work
Another issue I am having is using jupyter notebook instead of vs code. The problem is that it cannot connect to a kernel and I have found no solution to overcome this problem.

**Comment by connorjward** [comment]
These sorts of problems are very hard to fix without more information. When you say "the plots are always wrong" what do you mean? In what way are they wrong?

> Another issue I am having is using jupyter notebook instead of vs code. The problem is that it cannot connect to a kernel and I have found no solution to overcome this problem.

Did you launch Jupyter from a terminal with the `firedrake` virtual environment activated? Otherwise jupyter won't be able to find Firedrake.

**Reply by AntoTir** [reply]
Thank you for your answer. 
About the plots, it basically replicates always the first plot. For example if I change the colormap, the plot is not affected at all but only the colorbar on the right.
For jupyter yes, I activate the firedrake env and launch it from there

**Reply by connorjward** [reply]
> About the plots, it basically replicates always the first plot. For example if I change the colormap, the plot is not affected at all but only the colorbar on the right.

Bizarre. @JDBetteridge I notice you merged some matplotlib changes recently. Do you think there could be some issue with the latest version of matplotlib? @AntoTir what do you get as output from `$ pip list | grep matplotlib` and `$ pip list | grep jupyter`?

> For jupyter yes, I activate the firedrake env and launch it from there

I have just one guess here. Can you try launching with the command `$ python -m jupyter` instead of plain `$ jupyter`. It's feasible that the wrong Jupyter is getting launched even though the venv is activated. If I'm correct it may be necessary to `pip install jupyter`.

**Reply by AntoTir** [reply]
That's what I get as output
>matplotlib                3.9.0
>matplotlib-inline         0.1.7
-----
>jupyter_client            8.6.1
>jupyter_core              5.7.2

What really surprise me about the plot is that, the same code run on vscode and collab (where I install firedrake everytime) produces different output. Hence I have taught it was a problem of the installation on my PC even though I haven't received an error.

After pip install juoyter, the jupyter notebooks seem to work properly.
This let me think that maybe vs code is having problem with matplotlib

**Reply by connorjward** [reply]
Can you perhaps try running `pip install matplotlib==3.8.4` and see if that improves anything? There was a recent update of matplotlib which may be the source of your issues.

Glad to hear the other things are resolved.

**Reply by AntoTir** [reply]
Unfortunately vs code still gives problems. 

What about   >%matplotlib widget? I have simply commented it out since it gave me problems and I was not able to solve them looking at the solutions proposed in different forums

**Reply by connorjward** [reply]
I don't know what to suggest to fix your remaining issues. They do seem like they are problems with VS Code and matplotlib though so it may be constructive to try and use them in a separate virtual environment distinct from Firedrake.
